Followup changes in cconfig.c to utilize SETOPTIONS for config nodes

yetmorecode opened the pull request:
Changed lots of config nodes to use new config options with SETOPTIONS. Like:

```
user@linuxdev radare2-dev $ ./binr/radare2/radare2 --
 -- Wait a moment ...
[0x00000000]> e asm.emuskip=?
Skip metadata of given types in asm.emu:
d = data
c = code
s = string
f = format
m = magic
h = hide
C = comment
r = run
Concatenation of meta types encoded as characters.
Default is 'ds' to skip data and strings.
[0x00000000]> ev asm.emuskip
asm.emuskip = ds ; Skip metadata of given types in asm.emu [d = data, c = code, s = string, f = format, m = magic, h = hide, C = comment, r = run]
[0x00000000]> ev asm.arch
asm.arch = x86 ; Set the arch to be used by asm [6502, 8051, arc, arm.as, arm, arm.gnu, arm.winedbg, avr, bf, cr16, cris, dalvik, dcpu16, ebc, gb, h8300, hppa, i4004, i8080, java, lanai, lh5801, lm32, m68k, malbolge, mcs96, mips, mips.gnu, msp430, nios2, pic18c, ppc, ppc.gnu, riscv, rsp, sh, snes, sparc, sparc.gnu, spc700, sysz, tms320, tms320c64x, tricore, v810, v850, vax, wasm, ws, x86.as, x86, x86.nasm, x86.nz, x86.udis, xap, xcore, xtensa, z80]
[0x00000000]> ev asm.cpu
asm.cpu = x86 ; Set the kind of asm.arch cpu
[0x00000000]> e asm.arch=avr
[0x00000000]> ev asm.cpu
asm.cpu = ATmega8 ; Set the kind of asm.arch cpu [ATmega8, ATmega168, ATmega328p, ATmega32u4, ATmega1280, ATmega2560, ATmega48]
[0x00000000]>
```

yetmorecode commented:
Any tricks to identify related regressions? Im kinda lost in all the output from travis :/

radare commented:
your branch only broke one test. the others are also broken in master

<img width="612" alt="screen shot 2017-04-28 at 20 42 58" src="https://cloud.githubusercontent.com/assets/917142/25542547/54311db8-2c53-11e7-8b0b-11e161fb2c78.png">

yetmorecode commented:
Ok, that one is kind of expected. I print the config description in front of the option list now. I felt like it looks more helpful but I can also remove it again.

radare commented:
Remove it pls it will break the webui and other scripts. The output of ? Is expected to be only a newline separated list of options

> On 28 Apr 2017, at 21:05, yetmorecode <notifications@github.com> wrote:
> 
> Ok, that one is kind of expected. I print the config description in front of the option list now. I felt like it looks more helpful but I can also remove it again.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

yetmorecode commented:
Ok, removed it. Lets wait for travis.

FYI: There are a couple more config values not following that convention in `e var=?`, but there is also some unique documentation in there that would be lost if it was removed there. Like in:

```
[0x00000000]> e asm.arch=?
_dAe  _8_16      6502        LGPL3   6502/NES/C64/Tamagotchi/T-1000 CPU
_dA_  _8         8051        PD      8051 Intel CPU
_dA_  _16_32     arc         GPL3    Argonaut RISC Core
a___  _16_32_64  arm.as      LGPL3   as ARM Assembler (use ARM_AS environment)
adAe  _16_32_64  arm         BSD     Capstone ARM disassembler
_dA_  _16_32_64  arm.gnu     GPL3    Acorn RISC Machine CPU
_d__  _16_32     arm.winedbg LGPL2   WineDBG's ARM disassembler
adAe  _8_16      avr         GPL     AVR Atmel
adAe  _16_32_64  bf          LGPL3   Brainfuck
_dA_  _16        cr16        LGPL3   cr16 disassembly plugin
_dA_  _32        cris        GPL3    Axis Communications 32-bit embedded processor
adA_  _32_64     dalvik      LGPL3   AndroidVM Dalvik
ad__  _16        dcpu16      PD      Mojang's DCPU-16
_dA_  _32_64     ebc         LGPL3   EFI Bytecode
adAe  _16        gb          LGPL3   GameBoy(TM) (z80-like)
_dAe  _16        h8300       LGPL3   H8/300 disassembly plugin
_d__  _32        hppa        GPL3    HP PA-RISC
_dAe  _0         i4004       LGPL3   Intel 4004 microprocessor
_dA_  _8         i8080       BSD     Intel 8080 CPU
adA_  _32        java        Apache  Java bytecode
_d__  _32        lanai       GPL3    LANAI
_d__  _8         lh5801      LGPL3   SHARP LH5801 disassembler
_d__  _32        lm32        BSD     disassembly plugin for Lattice Micro 32 ISA
_d__  _16_32     m68k        BSD     Capstone M68K disassembler
_dA_  _32        malbolge    LGPL3   Malbolge Ternary VM
_d__  _16        mcs96       LGPL3   condrets car
adAe  _16_32_64  mips        BSD     Capstone MIPS disassembler
adAe  _32_64     mips.gnu    GPL3    MIPS CPU
_dA_  _16        msp430      LGPL3   msp430 disassembly plugin
_dA_  _32        nios2       GPL3    NIOS II Embedded Processor
_dAe  _8         pic18c      LGPL3   pic18c disassembler
_dAe  _32_64     ppc         BSD     Capstone PowerPC disassembler
_dA_  _32_64     ppc.gnu     GPL3    PowerPC
_dA_  _32_64     riscv       GPL     RISC-V
_dAe  _32        rsp         LGPL3   Reality Signal Processor
_dA_  _32        sh          GPL3    SuperH-4 CPU
_dA_  _8_16      snes        LGPL3   SuperNES CPU
_dAe  _32_64     sparc       BSD     Capstone SPARC disassembler
_dA_  _32_64     sparc.gnu   GPL3    Scalable Processor Architecture
_d__  _16        spc700      LGPL3   spc700, snes' sound-chip
_d__  _32        sysz        BSD     SystemZ CPU disassembler
_dA_  _32        tms320      LGPLv3  TMS320 DSP family
_dA_  _32        tms320c64x  BSD     Capstone TMS320c64x disassembler
_d__  _32        tricore     GPL3    Siemens TriCore CPU
_dAe  _32        v810        LGPL3   v810 disassembly plugin
_d__  _32        v850        LGPL3   v850 disassembly plugin
_dAe  _8_32      vax         GPL     VAX
_d__  _32        wasm        MIT     WebAssembly
_dA_  _32        ws          LGPL3   Whitespace esotheric VM
a___  _16_32_64  x86.as      LGPL3   Intel X86 GNU Assembler
_dAe  _16_32_64  x86         BSD     Capstone X86 disassembler
a___  _16_32_64  x86.nasm    LGPL3   X86 nasm assembler
a___  _16_32_64  x86.nz      LGPL3   x86 handmade assembler
_dAe  _16_32_64  x86.udis    BSD     udis86 x86-16,32,64
_dA_  _16        xap         PD      XAP4 RISC (CSR)
_dA_  _32        xcore       BSD     Capstone XCore disassembler
_dAe  _32        xtensa      GPL3    XTensa CPU
adA_  _8         z80         GPL     Zilog Z80
_d__  _32        propeller   LGPL3   propeller disassembly plugin
[0x00000000]> e asm.emuskip=?
d = data
c = code
s = string
f = format
m = magic
h = hide
C = comment
r = run
Concatenation of meta types encoded as characters.
Default is 'ds' to skip data and strings.
[0x00000000]>
```

IMO those should be kept and client like the web UI "encouraged" to rework with `evj` if things break

radare commented on libr/core/cconfig.c:
```diff
@@ -214,11 +222,21 @@ static int cb_analnoncode(void *user, void *data) {
+	RListIter *it;
+	r_list_purge (node->options);
+	r_list_foreach (core->anal->plugins, it, h) {
+		SETOPTIONS(node, h->name, NULL);
```
missing space before (

radare commented on libr/core/cconfig.c:
```diff
@@ -273,6 +291,49 @@ static int cb_asmassembler(void *user, void *data) {
+	RListIter *iter;
+	r_list_purge (node->options);
+	r_list_foreach (core->assembler->plugins, iter, h) {
+		SETOPTIONS(node, h->name, NULL);
```
missing space

radare commented on libr/core/cconfig.c:
```diff
@@ -485,18 +552,29 @@ static int cb_asmbits(void *user, void *data) {
+			argc = r_str_split (features, ',');
+			for (i = 0; i < argc; i++) {
+				const char *feature = r_str_word_get0 (features, i);
+				SETOPTIONS(node, feature, NULL);
```
missing space

radare commented on libr/core/cconfig.c:
```diff
@@ -538,10 +604,9 @@ static int cb_emustr(void *user, void *data) {
-				"'h': hide\n'C': comment\n'r': run\n" \
-				"(default is 'ds' to skip data and strings)\n");
+		print_node_options (node);
+		r_cons_printf ("Concatenation of meta types encoded as characters.\n" \
```
remove this

radare replied:
or put it with eprintf. but it looks noisy to me.or maybe you can add this with ??

radare replied:
if (node->value[1] == '?') { r_cons_printf("Concat..

yetmorecode replied:
`??` sounds good to me

radare commented on libr/core/cconfig.c:
```diff
@@ -1832,12 +1880,19 @@ R_API int r_core_config_init(RCore *core) {
 	//asm.os needs to be first, since other asm.* depend on it
-	SETCB("asm.os", R_SYS_OS, &cb_asmos, "Select operating system (kernel) (linux, darwin, w32,..)");
+	n = NODECB("asm.os", R_SYS_OS, &cb_asmos);
+	SETDESC(n, "Select operating system (kernel)");
```
add spaces before (

yetmorecode replied:
idk, I can do, but its the every pref missing those spaces here. Should I change them all?

radare replied:
Yes

radare commented on libr/core/cconfig.c:
```diff
@@ -1929,7 +1996,12 @@ R_API int r_core_config_init(RCore *core) {
-	SETCB("bin.strfilter", "", &cb_strfilter, "Filter strings (?:help, a:scii, e:mail, p:ath, u:rl, 8:utf8)");
+	n = NODECB("bin.strfilter", "", &cb_strfilter);
+	SETDESC(n, "Filter strings");
+	SETOPTIONS(n, "a = only alphanumeric printable", "8 = only strings with utf8 chars", NULL);
```
those parameters are a bit confusing

yetmorecode replied:
It's just what was printed before:

```
 -		eprintf ("Valid values for bin.strfilter:\n");
 -		eprintf ("a  only alphanumeric printable\n");		
 -		eprintf ("8  only strings with utf8 chars\n");		
 -		eprintf ("p  file/directory paths\n");		
 -		eprintf ("e  email-like addresses\n");		
 -		eprintf ("u  urls\n");		
 -		eprintf ("i  IPv4 address-like strings\n");		
 -		eprintf ("U  only uppercase strings\n");		
 -		eprintf ("f  format-strings\n");
```

I didnt change any config values at all

radare replied:
But the function parameters suggested different things to me than just a list of rows. Haviinh a single call to setoptions with all the options would be better

radare replied:
Also, bear in mind that we should be able to handle ? And ?? In all options to be consistent. Maybe we can show the verbose form with all this text and just an option per line without extra stuff in ?.

radare commented on libr/core/cconfig.c:
```diff
@@ -1832,12 +1880,19 @@ R_API int r_core_config_init(RCore *core) {
+	n = NODEICB("asm.midflags", 2, &cb_midflags);
+	SETDESC(n, "Realign disassembly if there is a flag in the middle of an instruction");
+	SETOPTIONS(n, "0 = do not show flag", "1 = show without realign", NULL);
+	SETOPTIONS(n, "2 = realign at middle flag", NULL);
```
why not just make this in a single call?

yetmorecode commented:
Ok, I added ??, fixed all the missing spaces and merged multiple calls into one. We unfortunately loose some help information in `evj`, but `?` is consistent that way.

radare commented:
lgtm
